marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/managers/EntitySearchManager.java

```diff
@@ -278,6 +290,126 @@ protected String buildSearchOptions(String query, SearchQuery searchQuery) {
         return sb.toString();
     }
 
+    public void exportById(String queryId, String fileType, Long limit, OutputStream out, HttpServletResponse response) {
+        JsonNode queryDocument = EntitySearchService.on(finalDataServiceClient).getSavedQuery(queryId);
+        exportByQuery(queryDocument, fileType, limit, out, response);
+    }
+
+    public void exportByQuery(JsonNode queryDocument, String fileType, Long limit, OutputStream out, HttpServletResponse response) {
+        if ("CSV".equals(fileType.toUpperCase())) {
+            prepareResponseHeader(response, CSV_CONTENT_TYPE, getFileNameForDownload(queryDocument, CSV_FILE_EXTENSION));
+            exportRows(queryDocument, limit, out);
+        }
+        else {
+            throw new DataHubException("Invalid file type: " + fileType);
+        }
+    }
+
+    public void exportRows(JsonNode queryDocument, Long limit, OutputStream out) {
+        QueryManager queryMgr = finalDatabaseClient.newQueryManager();
+        SearchQuery searchQuery = transformToSearchQuery(queryDocument);
+        StructuredQueryDefinition structuredQueryDefinition = buildQuery(queryMgr, searchQuery);
+
+        String structQuery = structuredQueryDefinition.serialize();
+        String searchQText = searchQuery.getQuery().getSearchText();
+        String queryOptions = finalDatabaseClient.newServerConfigManager().newQueryOptionsManager().readOptionsAs(QUERY_OPTIONS, Format.XML, String.class);
```

Comment:
I think we need to catch an exception here in case the options exist. They should exist - but if they don't, the user is going to get an error they likely won't understand and I'll probably get pinged with a message saying that Gradle is broken. 

So use a try/catch here to catch an exception if the options don't exist, and rethrow the error with a message of e.g. "Could not find search options: (name); system is not configured correctly, please contact a system administrator".

Reply:
Good catch. I'll catch the exception.